### PR TITLE
WIP: Adding Accessibility doc for Core WinForms

### LINKED ISSUE #13870: WinForms Accessibility Documentation
**WinForms Accessibility Documentation needs to be added**

- Need to add the description of new accessibility features, behavior changes and improvements added to .NET Core 3.0 WinForms.
- At this moment there is a documentation about WinForms Accessibility features compatibility in https://github.com/microsoft/dotnet/tree/master/Documentation/compatibility documents regarding changes in .NET Framework.

**WinForms Accessibility Documentation for .NET Core 3.0**

- There is a number of changes in WinForms Accessibility which are not documented yet.
- Accessibility changes in latest version of .NET Framework are documented but there are differences between .NET Framework 4.8 WinForms and .NET Core WinForms in this area.
- To describe the changes and compatibility between .NET Framework WinForms accessibility and .NET Core WinForms accessibility we need to add corresponding document.

### PR DESCRIPTION
**Pull request by M-Lipin**
## Summary

Adding Accessibility doc for Core WinForms which describes the changes made in WinForms accessibility in comparison with .NET Framework 4.8 https://github.com/microsoft/dotnet/tree/master/Documentation/compatibility

https://github.com/microsoft/dotnet/blob/master/Documentation/compatibility/winforms-accessibility-changes-48.md

Fixes #13870

### CODE REVIEW

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+title: "Windows Forms Accessibility Improvements"
+ms.date: "08/16/2019"
+helpviewer_keywords: 
+  - "Windows Forms accessibility"
+  - "accessibility"
+ms.assetid: 
+---
+# Accessibility improvements in Windows Forms controls for .NET Core
+
+## Scope
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]: Delete lines 11-17; they're based on the compatibility template, which isn't relevant here.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+Major
+
+## Version Introduced
+.NET Core 3.0
+
+## Change Description
+The Windows Forms is continuing to improve how it works with accessibility technologies to better support Windows Forms customers. These include the following changes:
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]:
```suggestion
Windows Forms is continuing to improve how it works with accessibility technologies to better support Windows Forms customers. These include the following changes:
```

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+- Changes in various areas of interaction with accessibility client applications including Narrator.
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]:
```suggestion
- Changes in various areas of interaction with accessibility client applications, including Narrator.
```

**Comment by rpetrusha** [comment]
There's not really a home for this article right now. Ultimately, it will have to go in the .NET Core Guide. I've created #13893 to do that.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+- Changes in the Accessible hierarchy (improving navigation through the UI Automation tree).
+- Changes in keyboard navigation.
+
+- [x] The quirks which were used in .NET Framework 4.7.1 - 4.8 to opt-in/out new accessibility features have been removed, so previously quirked behavior is now enabled by default.
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]:
```suggestion

```

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+### Recommended Action
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]:
```suggestion

```

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+__The changes which could be opted in or out in .NET Framework 4.7.1 - 4.8 are now enabled by default and cannot be opted out__
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]:
```suggestion
> [!IMPORTANT]
>  Accessibility changes made in .NET Framework 4.7.1 through .NET Framework 4.8 are included in .NET Core 3.0 and are enabled by default. The .NET Framework supported compatibility switches that allowed applications to opt out of the new accessibility behavior. .NET Core, on the other hand, does not support these settings and does not allow applications to opt out of accessibility behavior.
```

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  
+Applications now benefit from all the new accessibility features (introduced in .NET Framework 4.7.1 - 4.8) without additional configuration.
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]:
```suggestion
Starting with .NET Core 3.0, Windows Forms applications benefit from all the new accessibility features (introduced in .NET Framework 4.7.1 through .NET Framework 4.8 without additional configuration. These changes include the following.
```

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+__Improved ListBox Accessibility support__
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]: Each of these should be a level-2 head (\#\#)

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+- Enabled UIA support for ListBox control.
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]: Before the bulleted list, there should be a one-sentence introduction, like:

> On .NET Core, Windows Forms includes the following changes that affect the <xref:System.Windows.Forms.ListBox> control:

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+- Improved ListBox Accessibility support by adding the ScrollItem pattern to ListBox items and by enhancing the accessibility event raising and handling and Narrator navigation through the items (caps lock navigation is not correct and does not throw the navigation outside the control unintentionally).
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]:
```suggestion
- Improved ListBox accessibility support by adding the ScrollItem pattern to ListBox items and by enhancing the accessibility event raising and handling and Narrator navigation through the items (caps lock navigation is not correct and does not throw the navigation outside the control unintentionally).
```
But the parenthetical expression seems like a non sequitur. Is it an exception? A qualification?

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+- Improved High Contrast themes support and made the control element more contrasted.
+
+__StringCollectionEditor Accessibility support__
+- Added StringCollectionEditor with improved accessibility support.
+
+__UI automation support for DataGridView, PropertyGrid, ListBox, ComboBox, ToolStrip and other controls__
+Note: UI automation support is enabled for controls in runtime but is not used in design time.
+For an overview of UI automation, see the [UI Automation Overview](https://docs.microsoft.com/dotnet/framework/ui-automation/ui-automation-overview).
+
+
```
Comment (rpetrusha) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]: Lines 65 through 70 can be deleted. Optionally, the remaining commented-out list can be deleted.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
I've left a number of comments, @M-Lipin. This article more closely needs to resemble [What's new in accessibility in .NET Framework](https://docs.microsoft.com/en-us/dotnet/framework/whats-new/whats-new-in-accessibility) than the old compatibility items in the Microsoft/dotnet repo. I'll do another review once these changes are made.

**Comment by rpetrusha** [comment]
//cc @Thraka

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
+---
+title: "Windows Forms Accessibility Improvements"
+ms.date: "08/16/2019"
+helpviewer_keywords: 
+  - "Windows Forms accessibility"
+  - "accessibility"
+ms.assetid:
```
[the remark this answers is not in the document]
Comment (RussKie) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: `author: "M-Lipin"`

**Comment by RussKie** [comment]
It could be a good idea to wrap control names into `` ` ``, e.g. ``...`ListBox` control...``

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
+---
+title: "Windows Forms Accessibility Improvements"
+ms.date: "08/16/2019"
+helpviewer_keywords: 
+  - "Windows Forms accessibility"
+  - "accessibility"
+ms.assetid: 
+---
+# Accessibility improvements in Windows Forms controls for .NET Core
```
[the remark this answers is not in the document]
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 1 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 2 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 3 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 7 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 8 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 9 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 10 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 11 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed, thanks.

**Comment by Thraka** [comment]
@M-Lipin Hello. There are a few changes you should accept as suggested by Ron. Also, we have a linter running to ensure that the markdown content we accept is well formed. Can you please address those? If you're using VSCode you should be getting the linter rules if you open the repo root. Also, you can see the lint results in the **Checks** tab at the top.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
+---
+title: "Windows Forms Accessibility Improvements"
+ms.date: "08/16/2019"
```
Comment (Thraka) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]: - Can you add a `description: ` entry too, it should be around 100 characters long if possible. It should not repeat what the first paragraph in the article says.

- Also the date can be updated 😄

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+On .NET Core, Windows Forms includes the following changes that affect the xref:System.Windows.Forms.ListBox control:
+- Enabled UIA support for ListBox control.
+- Improved ListBox accessibility support by adding the ScrollItem pattern to ListBox items and by enhancing the accessibility event raising and handling and Narrator navigation through the items (caps lock navigation is not correct and does not throw the navigation outside the control unintentionally).
+
+## Improved CheckedListBox Accessibility support
+On .NET Core, Windows Forms includes the following changes that affect the xref:System.Windows.Forms.CheckedListBox control:
+- Corrected CheckedListBox Bounds provided by accessibility properties for entries. Improved overall ListBox and CheckedListBox accessibility (corrected pproperty values and event model).
+
+## Improved ComboBox Accessibility support
+On .NET Core, Windows Forms includes the following changes that affect the xref:System.Windows.Forms.ComboBox control:
```
Comment (Thraka) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]: These refs need to be fixed, they are actually XML entries:

```suggestion
On .NET Core, Windows Forms includes the following changes that affect the <xref:System.Windows.Forms.ComboBox> control:
```

You can use this format for all controls referenced in this article.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 14 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Thank your for the review! Added description, fixed date.

File: docs/framework/winforms/windows-forms-accessibility-improvements.md (hunk 15 above)
Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed this, thanks!

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+Starting with .NET Core 3.0, Windows Forms applications benefit from all the new accessibility features (introduced in .NET Framework 4.7.1 - 4.8) without additional configuration.
+
+## Improved ListBox Accessibility support
+On .NET Core, Windows Forms includes the following changes that affect the <xref:System.Windows.Forms.ListBox control>:
+- Enabled UIA support for ListBox control.
+- Improved ListBox accessibility support by adding the ScrollItem pattern to ListBox items and by enhancing the accessibility event raising and handling and Narrator navigation through the items (caps lock navigation is not correct and does not throw the navigation outside the control unintentionally).
+
+## Improved CheckedListBox Accessibility support
+On .NET Core, Windows Forms includes the following changes that affect the <xref:System.Windows.Forms.CheckedListBox control>:
+- Corrected CheckedListBox Bounds provided by accessibility properties for entries. Improved overall ListBox and CheckedListBox accessibility (corrected pproperty values and event model).
```
Comment (RussKie) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]:
```suggestion
- Corrected CheckedListBox Bounds provided by accessibility properties for entries. 
- Improved overall ListBox and CheckedListBox accessibility (corrected pproperty values and event model).
```

File: docs/framework/winforms/windows-forms-accessibility-improvements.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+
+## Improved PropertyGrid Accessibility support
+On .NET Core, Windows Forms includes the following changes that affect the <xref:System.Windows.Forms.PropertyGrid control>:
+- Corrected PropertyGrid Bounds provided by accessibility properties for grid entries, improved performance of bounding rectangle calculation. For now all accessibility bounds are represented correctly taking into account the bounds of entire control, its viewport and scrolls.
+- Corrected accessible names and descriptions of subcontrols to not include control type names and to avoid double announcement for control type names.
+
+## Improved ToolStrip Accessibility support
+On .NET Core, Windows Forms includes the following changes that affect the <xref:System.Windows.Forms.ToolStrip control>:
+- Improved navigation through ToolStrip items. Corrected ToolStrip and Menu shift-tab navigation (back-looping the menu items when shift-tab up-arrow is pressed - navigate to the bottom menu element)
+- Improved Menu accessible navigation, corrected menu accessible control types for submenus to make submenus of type 'Menu' instead of 'MenuItem'.
```
Comment (RussKie) [inline docs/framework/winforms/windows-forms-accessibility-improvements.md]: We have removed `Menu` and `MenuItem` types in 3.1. 
Please verify this sentence is still applicable

Comment (Thraka) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: What is there in its place?

Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: At this moment there are only ToolStrip, MenuStrip and StatusStrip controls. These controls fully cover the menu bar/tool bar functionality and there is no changes in the UI.

Comment (M-Lipin) [reply docs/framework/winforms/windows-forms-accessibility-improvements.md]: Fixed @RussKie 's review point. Thanks!

**Comment by Thraka** [comment]
@M-Lipin Hello. Sorry about losing sight of this pull request over the holiday season. We had previously implemented a linter to ensure we have constant structure in the markdown used. There are a few changes that need to happen. Would you prefer it if I edited your doc to fix these problems and ultimately get this pushed through? Or would you rather fix them yourself?

Let me know.

**Comment by M-Lipin** [comment]
Hello @Thraka sorry for the late response. I can fix the document issues. Could you please point me to the issues/requirements or sample how to fix?

**Comment by tdykstra** [comment]
@M-Lipin here are the markdownlint error messages that @Thraka was referring to.  Fixing them just involves making sure there's a blank line before and after a list.

docs/framework/winforms/windows-forms-accessibility-improvements.md:13 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Changes in various areas of ..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:26 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Enabled UIA support for List..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:31 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Corrected CheckedListBox Bou..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:36 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Updated the process of getti..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:40 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Corrected DataGridView Bound..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:47 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Corrected PropertyGrid Bound..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:52 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Improved navigation through ..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:57 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Improved accessible navigati..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:65 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Added UI Automation server p..."]
docs/framework/winforms/windows-forms-accessibility-improvements.md:72 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Added ability to announce a ..."]

**Comment by Thraka** [comment]
@M-Lipin I'm starting on some WinForms planning for .NET Core content. I can jump in here and fix this up for you today. Is that OK?

**Comment by M-Lipin** [comment]
> @M-Lipin I'm starting on some WinForms planning for .NET Core content. I can jump in here and fix this up for you today. Is that OK?

It would be great. Thank you!

**Comment by Thraka** [comment]
Completing this today so I can continue with the desktop guide winforms planning.

**Review by Thraka — APPROVED** [review]

**Pull request merged**